lib/mayaUsd/render/vp2RenderDelegate/mesh.cpp

```diff
@@ -544,42 +544,77 @@ void HdVP2Mesh::_PrepareSharedVertexBuffers(
                 = _getInfo(_meshSharedData->_primvarInfo, HdVP2Tokens->displayColorAndOpacity);
         }
 
-        if (!colorAndOpacityInfo->_buffer) {
-            const MHWRender::MVertexBufferDescriptor vbDesc(
-                "", MHWRender::MGeometry::kColor, MHWRender::MGeometry::kFloat, 4);
+        if (colorInterp == HdInterpolationInstance || alphaInterp == HdInterpolationInstance) {
+            TF_VERIFY(!GetInstancerId().IsEmpty());
+            VtIntArray instanceIndices = delegate->GetInstanceIndices(GetInstancerId(), GetId());
+            size_t     numInstances = instanceIndices.size();
+            colorAndOpacityInfo->_extraInstanceData.setLength(
+                numInstances * kNumColorChannels); // the data is a vec4
+            void* bufferData = &colorAndOpacityInfo->_extraInstanceData[0];
+
+            size_t alphaChannelOffset = 3;
+            for (size_t instance = 0; instance < numInstances; instance++) {
+                int      index = instanceIndices[instance];
+                GfVec3f* color = reinterpret_cast<GfVec3f*>(
+                    reinterpret_cast<float*>(&static_cast<GfVec4f*>(bufferData)[instance]));
+                float* alpha
+                    = reinterpret_cast<float*>(&static_cast<GfVec4f*>(bufferData)[instance])
+                    + alphaChannelOffset;
```

Comment:
Yuck! The clang-format looks ugly here, causing different formats for different lines...

Reply:
I guess we can make it look better if we reduce the indentation on the left side, so it doesn't have to split lines. Maybe some of this can be moved into a separate function to make it more readable?